shell/common/process_util.cc

```diff
@@ -40,4 +40,11 @@ bool IsRendererProcess() {
   return process_type == switches::kRendererProcess;
 }
 
+bool IsUtilityProcess() {
+  auto* command_line = base::CommandLine::ForCurrentProcess();
+  std::string process_type =
```

Comment:
Unrelated to the landing of this PR but I would note that depending how often this gets checked there could be a perf hit compared to previous implementations.  Should we make `process_type` a static `base::NoDestructor` and cache it for the lifetime of the process, to avoid pulling it from `CommandLine` every time?

Reply:
I can do that in a separate PR

Reply:
If we cache the bool instead of the string, e.g. with an `absl::optional<bool>`, we can make the function a little faster and also avoid the `NoDestructor` hit

Reply:
or let's just do it here to simplify the backports